### PR TITLE
WIP: Sphinx documentation for Rubi

### PR DESCRIPTION
**Pull request by Abdullahjavednesar**

### CODE REVIEW

**Comment by Abdullahjavednesar** [comment]
@asmeurer, @jksuom, @certik, please review this.
It throws ```ImportError: No module named 'sympy.integrals.rubi'```.

File: doc/src/modules/integrals/rubi/about.rst [hunk 1]
```diff
@@ -0,0 +1,15 @@
+About Rule Based Integrator
+===========================
+
+About it
+--------
+
+A Rule based integrator nicknamed Rubi is module in SymPy, Integrals. It is an implementation of more than 6000 rules to cover a wide variety of
```
Comment (gxyd) [inline doc/src/modules/integrals/rubi/about.rst]: Better written as `RUle Based Integrator` may be.

File: sympy/integrals/rubi/utility_function.py [hunk 2]
```diff
@@ -258,7 +261,7 @@ def IntegersQ(*var):
     return all(IntegerQ(i) for i in var)
 
 def ComplexNumberQ(*var):
```
Comment (gxyd) [inline sympy/integrals/rubi/utility_function.py]: `*var` -> `*vars`.

**Comment by vishalg2235** [comment]
I would like to work on this. Is this complete @Abdullahjavednesar ?

File: doc/src/modules/integrals/rubi/about.rst [hunk 3]
```diff
@@ -0,0 +1,15 @@
+About Rule Based Integrator
+===========================
+
+About it
+--------
+
+A Rule based integrator nicknamed Rubi is module in SymPy, Integrals. It is an implementation of more than 6000 rules to cover a wide variety of 
+indefinite integration. Rubi utilizes a set of well defined rules which makes it smart to present the results in a more symmetric and simplified
```
Comment (gxyd) [inline doc/src/modules/integrals/rubi/about.rst]: It doesn't sound good "cover a wide variety of indefinite integration."

File: doc/src/modules/integrals/rubi/about.rst [hunk 4]
```diff
@@ -0,0 +1,15 @@
+About Rule Based Integrator
+===========================
+
+About it
+--------
+
+A Rule based integrator nicknamed Rubi is module in SymPy, Integrals. It is an implementation of more than 6000 rules to cover a wide variety of 
+indefinite integration. Rubi utilizes a set of well defined rules which makes it smart to present the results in a more symmetric and simplified 
+manner. The Pattern Matcher (MatchPy) along with transformation rules plays an important role in making Rubi more robust and fast. As it works on
```
Comment (gxyd) [inline doc/src/modules/integrals/rubi/about.rst]: Some reference to MatchPy would be good.

File: sympy/integrals/rubi/utility_function.py [hunk 5]
```diff
@@ -258,7 +261,7 @@ def IntegersQ(*var):
...
-    """
+    r"""
```
Comment (gxyd) [inline sympy/integrals/rubi/utility_function.py]: Why use `r` here?
Python documentation says:
```
Use r"""raw triple double quotes""" if you use any backslashes in your
docstrings.
```
See https://www.python.org/dev/peps/pep-0257/#what-is-a-docstring

File: doc/src/modules/integrals/rubi/about.rst (hunk 1 above)
Comment (gxyd) [inline doc/src/modules/integrals/rubi/about.rst]: Better use `module in sympy.integrals`?

**Review by gxyd — CHANGES_REQUESTED** [review]

**Comment by Abdullahjavednesar** [comment]
@vishalg2235, @gxyd has requested changes I think you should fix those and hopefully then we are done.

File: doc/src/modules/integrals/rubi/about.rst (hunk 3 above)
Comment (vishalg2235) [reply doc/src/modules/integrals/rubi/about.rst]: what should I write instead of this ?

File: sympy/integrals/rubi/utility_function.py (hunk 2 above)
Comment (vishalg2235) [reply sympy/integrals/rubi/utility_function.py]: Many functions has used "*var". So should I change all of them to "*vars" .And, also in function body.

Comment (gxyd) [reply sympy/integrals/rubi/utility_function.py]: No, please ignore making this change.

**Comment by czgdp1807** [comment]
@Abdullahjavednesar Is the work on this PR still on?

**Comment by czgdp1807** [comment]
Closing this in favour of https://github.com/sympy/sympy/pull/17892

**Pull request closed without merging**